pkg/controller/operators/catalog/operator.go

```diff
@@ -1352,8 +1352,15 @@ type UnpackedBundleReference struct {
 	Properties             string `json:"properties"`
 }
 
-// unpackBundles makes one walk through the bundlelookups and attempts to progress them
-func (o *Operator) unpackBundles(plan *v1alpha1.InstallPlan) (bool, *v1alpha1.InstallPlan, error) {
+/* unpackBundles makes one walk through the bundlelookups and attempts to progress them
+Returns:
+    unpacked: bool - If the bundle was successfully unpacked
+    out:      *v1alpha1.InstallPlan - the resulting installPlan
+	isFatal:  bool - if the resulting error was fatal (not retryable)
+	error:    error
+*/
+
+func (o *Operator) unpackBundles(plan *v1alpha1.InstallPlan) (bool, *v1alpha1.InstallPlan, bool, error) {
```

Comment:
Great start! I would now remove the errorIsFatal boolean from the return. Then the callers of unpackBundles can just us olmerrors.IsRetryable